front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionSelector.tsx

```diff
@@ -25,6 +30,9 @@ export default function PowerRestrictionSelector() {
   const [powerRestrictions, setPowerRestrictions] = useState<selectorOption[] | undefined>();
   const [getRollingStockById] = osrdEditoastApi.endpoints.getLightRollingStockById.useLazyQuery({});
   const [getPathFindingById] = osrdMiddlewareApi.endpoints.getPathfindingById.useLazyQuery({});
+  const [getPathFindingByIdCatenaries] =
+    osrdEditoastApi.useLazyGetPathfindingByPathIdCatenariesQuery();
+  const [getRollingStockWithCurves] = osrdEditoastApi.useLazyGetRollingStockByIdQuery();
```

Comment:
Fetch the pathFinding and the rollingStock outside of the useEffect or functions (ideally, we could get them as props of the component)

Reply:
- Sadly the parent component doesn't use rollingstock. I could make a state in the parent to hydrate both rollingstock and powerrestriction components but not sure I should do this kind of refacto in this issue ?
- Same for pathfinding with catenaries which is specificly used in this component
- The normal pathfinding is used in the parent but to pass it at props would involved some refacto in the parent (it's currently fetched with a lazyquery). So again, not sure if I should do the refacto in this issue ?
- I was fetching 2 times a normal rollingstock and the light rollingstock from Théo (as you pointed it during our meeting at the office I think) so I merged in only one query of the "not light" rollingstock
- On the other hand, I was able to remove all fetch from functions and useEffect with a standard query + skip parameter as we saw together (this component exists only if pathfinding and rollingstock exist so I think it's ok)

Reply:
OK nice, no need to do the refacto with the props then ! :)